src/options/getRepoOptions.ts

```diff
@@ -0,0 +1,40 @@
+import { cosmiconfigSync } from 'cosmiconfig';
+import { getDefaultRemoteBranch } from 'workspace-tools';
+import { RepoOptions, CliOptions } from '../types/BeachballOptions';
+
+export function getRepoOptions(cliOptions: CliOptions): RepoOptions {
+  let repoOptions: RepoOptions | null;
+  if (cliOptions.configPath) {
+    repoOptions = tryLoadConfig(cliOptions.configPath);
+    if (!repoOptions) {
+      console.error(`Config file "${cliOptions.configPath}" could not be loaded`);
+      process.exit(1);
+    }
+  } else {
+    repoOptions = trySearchConfig();
+  }
+  repoOptions = repoOptions || ({} as RepoOptions);
+
+  if (!cliOptions.branch) {
+    // Add or fix up the branch in repoOptions if it's not specified in cliOptions (which takes precedence)
+    if (repoOptions.branch && !repoOptions.branch.includes('/')) {
+      repoOptions.branch = getDefaultRemoteBranch(repoOptions.branch, cliOptions.path);
+    } else if (!repoOptions.branch) {
+      repoOptions.branch = getDefaultRemoteBranch('master', cliOptions.path);
+    }
+  }
+
+  return repoOptions;
+}
+
+function tryLoadConfig(configPath: string): RepoOptions | null {
+  const configExplorer = cosmiconfigSync('beachball');
```

Comment:
**thought:** this will throw Error if no config is found. are we sure we don't wanna handle it ?

Reply:
This isn't a new file, just renamed, so I'm not necessarily looking to fix up things beyond the remote selection. For this specific issue, I don't see anything in the cosmiconfig docs about throwing if not found, and testing it out locally it doesn't appear to throw. https://www.npmjs.com/package/cosmiconfig